mm2src/coins/lightning.rs

```diff
@@ -376,12 +614,52 @@ impl SwapOps for LightningCoin {
         _min_block_number: u64,
         _uuid: &[u8],
     ) -> Box<dyn Future<Item = (), Error = String> + Send> {
-        unimplemented!()
+        Box::new(futures01::future::ok(()))
     }
 
     fn validate_maker_payment(&self, _input: ValidatePaymentInput) -> ValidatePaymentFut<()> { unimplemented!() }
 
-    fn validate_taker_payment(&self, _input: ValidatePaymentInput) -> ValidatePaymentFut<()> { unimplemented!() }
+    fn validate_taker_payment(&self, input: ValidatePaymentInput) -> ValidatePaymentFut<()> {
+        let payment_hash = try_f!(payment_hash_from_slice(&input.payment_tx)
+            .map_to_mm(|e| ValidatePaymentError::TxDeserializationError(e.to_string())));
```

Comment:
> Probably there should be a callback called from maker_swap and taker_swap that the swap is failed, and lightning will free the corresponding htlc

I don't get what you mean by a `callback`? isn't the error returned in the below code enough?
https://github.com/KomodoPlatform/atomicDEX-API/blob/8b515c860faf640e04438c694561bdc8b9093ee4/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L918-L925

Reply:
Maybe I can edit the error message to include this information too. What do you think @sergeyboyko0791?

Reply:
I meant that the following code
```rust
// Free the htlc to allow for this inbound liquidity to be used for other inbound payments
coin.channel_manager.fail_htlc_backwards(&payment_hash);
payment.status = HTLCStatus::Failed;
drop_mutability!(payment);
coin.db
    .add_or_update_payment_in_db(payment)
    .await
    .error_log_with_msg("Unable to update payment information in DB!");
return MmError::err(ValidatePaymentError::WrongPaymentTx(format!(
    "Provided payment {} amount {:?} doesn't match required amount {}",
    payment_hex, amount_sent, amt_msat
)));
```
might be used on any error occurred at the `validate_taker_payment`, `send_maker_payment` etc as it changes the state of the `payment` to `HTLCStatus::Failed`. Does this even make sense?

Reply:
> Does this even make sense?

Yes, I understand now, thanks for the clarification :)
Let's make this part of the next sprint since I need to think about it in more details. I will add this comment to the lightning checklist.